src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.skia.cs

```diff
@@ -321,31 +314,98 @@ private static Rect CreateImageLayer(Compositor compositor, CompositeDisposable
 		/// <summary>
 		/// Creates a rounded-rectangle path from the nominated bounds and corner radius.
 		/// </summary>
-		private static CompositionPath GetRoundedPath(CornerRadius cornerRadius, Rect area, SkiaGeometrySource2D geometrySource = null)
+		private static CompositionPath GetRoundedPath(SKRect area, SKPoint[] radii, SkiaGeometrySource2D geometrySource = null)
 		{
 			geometrySource ??= new SkiaGeometrySource2D();
 			var geometry = geometrySource.Geometry;
 
 			// How ArcTo works:
 			// http://www.twistedape.me.uk/blog/2013/09/23/what-arctopointdoes/
 
-			geometry.MoveTo((float)area.GetMidX(), (float)area.Y);
-			geometry.ArcTo((float)area.Right, (float)area.Top, (float)area.Right, (float)area.GetMidY(), (float)cornerRadius.TopRight);
-			geometry.ArcTo((float)area.Right, (float)area.Bottom, (float)area.GetMidX(), (float)area.Bottom, (float)cornerRadius.BottomRight);
-			geometry.ArcTo((float)area.Left, (float)area.Bottom, (float)area.Left, (float)area.GetMidY(), (float)cornerRadius.BottomLeft);
-			geometry.ArcTo((float)area.Left, (float)area.Top, (float)area.GetMidX(), (float)area.Top, (float)cornerRadius.TopLeft);
-
+			var roundRect = new SKRoundRect();
+			roundRect.SetRectRadii(area, radii);
+			geometry.AddRoundRect(roundRect);
 			geometry.Close();
 
 			return new CompositionPath(geometrySource);
 		}
 
-		private static CompositionPath GetRoundedRect(CornerRadius cornerRadius, CornerRadius innerCornerRadius, Rect area, Rect insetArea)
+		internal static SKPoint[] GetOuterRadii(CornerRadius cornerRadius, Thickness borderThickness, float maxHorizontalRadius, float maxVerticalRadius)
```

Comment:
It'd probable be causing less allocations if the return value was a record struct (the size is always 4).

Reply:
@jeromelaban Agreed, but the SkiaSharp `SetRectRadii` method accepts only an array with exactly 4 elements. It would be nice to reuse the array but array pool can return larger arrays, which would cause exception.

Reply:
@jeromelaban  As I think about it, it should be safe to have two static fields for the arrays, because the method can only run on UI thread, so there should be no race condition, no?

Reply:
Indeed, because it's the UI Thread, it should be fine.